near-contract-standards/src/event.rs

```diff
@@ -0,0 +1,283 @@
+use std::borrow::Cow;
+
+use serde::{Deserialize, Serialize};
+use serde_with::skip_serializing_none;
+
+#[derive(Serialize, Debug)]
+#[serde(tag = "standard")]
+#[serde(rename_all = "snake_case")]
+pub enum NearEvent<'a> {
+    #[serde(borrow)]
+    Nep171(Nep171Event<'a>),
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct Nep171Event<'a> {
+    pub version: &'static str,
+    #[serde(flatten)]
+    #[serde(borrow)]
+    pub event_kind: Nep171EventKind<'a>,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(tag = "event", content = "data")]
+#[serde(rename_all = "snake_case")]
+#[allow(clippy::enum_variant_names)]
+pub enum Nep171EventKind<'a> {
+    #[serde(borrow)]
+    NftMint(Vec<NftMintData<'a>>),
+    #[serde(borrow)]
+    NftTransfer(Vec<NftTransferData<'a>>),
+    #[serde(borrow)]
+    NftBurn(Vec<NftBurnData<'a>>),
+}
+
+#[skip_serializing_none]
+#[derive(Serialize, Deserialize, Debug)]
+pub struct NftMintData<'a> {
+    #[serde(borrow)]
+    pub owner_id: Cow<'a, str>,
```

Comment:
Maybe worth putting in the comments somewhere in this file about why the optimization with `Cow`'s are needed, or linking back to this PR with the discussions could work too